test/WebJobs.Script.Tests/StaticAnalysisTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Controllers;
+using Microsoft.Azure.WebJobs.Script.WebHost.Filters;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests
+{
+    public class StaticAnalysisTests
+    {
+        /// <summary>
+        /// This test exists to ensure we don't add any host APIs that return secrets via GET,
+        /// unless they're marked with the <see cref="ResourceContainsSecretsAttribute"/>.
```

Comment:
The approach taken years ago when ArmExtensionResourceFilter was added relied on explicit marking of GET actions that return secrets. That requires anyone adding new GET actions to know what to do, and not to forget. This regression test ensures that mistakes can't be made.